docs/virtual_machines/net_binding_plugins/macvtap.md

```diff
@@ -0,0 +1,218 @@
+And in detail:
+
+### Macvtap CNI and DP deployment on nodes
+
+To simplify the procedure, use the
+[Cluster Network Addons Operator](https://github.com/kubevirt/cluster-network-addons-operator#macvtap)
+to deploy and configure the macvtap components in your cluster.
+
+The aforementioned operator effectively deploys the
+[macvtap](https://github.com/kubevirt/macvtap-cni) cni and device plugin.
+
+### Expose node interface to the macvtap device plugin
+
+There are two different alternatives to configure which host interfaces get
+exposed to the user, enabling them to create macvtap interfaces on top of:
+
+- select the host interfaces: indicates which host interfaces are exposed.
+- expose all interfaces: all interfaces of all hosts are exposed.
+
+Both options are configured via the `macvtap-deviceplugin-config` ConfigMap,
+and more information on how to configure it can be found in the
+[macvtap-cni](https://github.com/kubevirt/macvtap-cni#deployment) repo.
+
+This is a minimal example, in which the `eth0` interface of the Kubernetes
+nodes is exposed, via the `master` attribute.
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: macvtap-deviceplugin-config
+data:
+  DP_MACVTAP_CONF: |
+    [
+        {
+            "name"     : "dataplane",
+            "master"   : "eth0",
+            "mode"     : "bridge",
+            "capacity" : 50
+        },
+    ]
+```
+
+This step can be omitted, since the default configuration of the aforementioned
+`ConfigMap` is to expose all host interfaces (which is represented by the
+following configuration):
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: macvtap-deviceplugin-config
+data:
+  DP_MACVTAP_CONF: '[]'
+```
+
+### Macvtap NetworkAttachmentDefinition
+
+The configuration needed for a macvtap network attachment can be minimalistic:
+
+```yaml
+kind: NetworkAttachmentDefinition
+apiVersion: k8s.cni.cncf.io/v1
+metadata:
+  name: macvtapnetwork
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: macvtap.network.kubevirt.io/eth0
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "macvtapnetwork",
+      "type": "macvtap",
+      "mtu": 1500
+    }'
+```
+
+The object should be created in a "default" namespace where all other namespaces
+can access, or, in the same namespace the VMs reside in.
+
+The requested `k8s.v1.cni.cncf.io/resourceName` annotation must point to an
+exposed host interface (via the `master` attribute, on the
+`macvtap-deviceplugin-config` `ConfigMap`).
+
+### Feature Gate
+If not already set, add the `NetworkBindingPlugins` FG.
+```
+kubectl patch kubevirts -n kubevirt kubevirt --type=json -p='[{"op": "add", "path": "/spec/configuration/developerConfiguration/featureGates/-",   "value": "NetworkBindingPlugins"}]'
+```
+
+> **Note**: The specific macvtap plugin has no FG by its own. It is up to the cluster
+> admin to decide if the plugin is to be available in the cluster.
+> The macvtap binding is still in evaluation, use it with care.
```

Comment:
There is one to protect the core macvtap binding that exists today.

With the plugins, there is no longer a need for it. It is up to the cluster-admin if to register it for usage in the cluster and up to the VM-admin if to use it in the VM spec.

To clarify, the original core macvtap binding is not yet removed. It will be deprecated with a notification sent soon.

Reply:
OK OK, makes sense. 

Thanks. Do you think it would make sense to explain this in https://kubevirt.io/user-guide/virtual_machines/interfaces_and_networks/#macvtap ?

Reply:
disreguard this comment. I am unsure it would be helpful to the user.